platform/darwin/src/collator.mm

```diff
@@ -0,0 +1,63 @@
+#include <mbgl/style/expression/collator.hpp>
+
+#import <Foundation/Foundation.h>
+
+namespace mbgl {
+namespace style {
+namespace expression {
+    
+class Collator::Impl {
+public:
+    Impl(bool caseSensitive, bool diacriticSensitive, optional<std::string> locale_)
+    : options(caseSensitive ? NSCaseInsensitiveSearch : 0 |
+              diacriticSensitive ? NSDiacriticInsensitiveSearch : 0)
+    , locale(locale_ ?
+                [[NSLocale alloc] initWithLocaleIdentifier:[NSString stringWithUTF8String:locale_->c_str()]] :
+                [NSLocale currentLocale])
+    {}
+    
+    bool operator==(const Impl& other) const {
+        return options == other.options &&
+            [[locale localeIdentifier] isEqualToString:[other.locale localeIdentifier]];
+    }
+    
+    int compare(const std::string& lhs, const std::string& rhs) const {
+        NSString* nsLhs = [NSString stringWithUTF8String:lhs.c_str()];
```

Comment:
Nit: you can write `@(lhs.c_str())` instead of calling `-[NSString stringWithUTF8String:]` explicitly.

Reply:
Ok